src/fmeval/model_runners/util.py

```diff
@@ -9,12 +9,32 @@
 import botocore.config
 import sagemaker
 
-from fmeval.constants import SAGEMAKER_SERVICE_ENDPOINT_URL, SAGEMAKER_RUNTIME_ENDPOINT_URL
+from fmeval.constants import SAGEMAKER_SERVICE_ENDPOINT_URL, SAGEMAKER_RUNTIME_ENDPOINT_URL, DISABLE_FMEVAL_TELEMETRY
+from fmeval.util import get_fmeval_package_version
+from sagemaker.user_agent import determine_prefix
 from mypy_boto3_bedrock.client import BedrockClient
 
+
 logger = logging.getLogger(__name__)
 
 
+def get_user_agent_extra() -> str:
+    """
+    :return: A string to be used as the user_agent_extra parameter in a botocore config.
+    """
+    # Obtain user-agent headers for information such as SageMaker notebook instance type and SageMaker Studio app type.
```

Comment:
where is this information coming from? `determine_prefix`? If yes, then can we make it more apparent in docstrings. 

And `determine_prefix` will give us all the headers. right? Not just the ones added by sagemaker_python_sdk.

Reply:
`determine_prefix` is a [util function imported from the SageMaker Python SDK](https://github.com/aws/sagemaker-python-sdk/blob/2a52478e7fd185f197e944568173c58bd2495d78/src/sagemaker/user_agent.py#L66C5-L93). It returns only python SDK-related headers, and not all headers.

Our code is using the `user_agent_extra` parameter to pass the SDK headers and our own `fmeval` header as "extras", which get added to whatever headers are already present in the user agent.